lib/Doctrine/DBAL/Platforms/SQLServer2012Platform.php

```diff
@@ -120,9 +120,18 @@ protected function doModifyLimitQuery($query, $limit, $offset = null)
         if ($orderByPos === false
             || substr_count($query, "(", $orderByPos) - substr_count($query, ")", $orderByPos)
         ) {
-            // In another DBMS, we could do ORDER BY 0, but SQL Server gets angry if you use constant expressions in
-            // the order by list.
-            $query .= " ORDER BY (SELECT 0)";
+            if (stripos($query, 'SELECT DISTINCT') === 0) {
+                // SQL Server won't let us order by a non-selected column in a DISTINCT query,
+                // so we have to do this madness. This says, order by the first column in the
```

Comment:
I'm fine with this PR as is for now. We should add the integration tests for the limit/offset modifications in seperate PR anyways.